LeaderboardBackend/Services/IUserService.cs

```diff
@@ -18,7 +18,7 @@ public interface IUserService
     // TODO: Convert return sig to Task<GetUserResult>
     Task<User?> GetUserByName(string name);
     Task<User?> GetUserByNameAndEmail(string name, string email);
-    Task<ListResult<User>> ListUsers(Page page, HashSet<UserRole> roles);
+    Task<ListResult<User>> ListUsers(Page page, UserRole roles);
```

Comment:
Rename for consistency:
```suggestion
    Task<ListResult<User>> ListUsers(Page page, UserRole roleFlags);
```